group/secp256k1/scalar_test.go

```diff
@@ -0,0 +1,127 @@
+package secp256k1
+
+import (
+	"encoding/hex"
+	"math/big"
```

Comment:
group/nist/curve.go uses group/mod to do scalar modular math, why did you decide not to use it? I have to say, I'm not totally clear on the difference, what would be the benefits, etc. But I generally understand that the SECG group's curves are like the NIST curves, but with constants that are more transparent ("no numbers up my sleeves"), and so I'm wondering why the implementation in this directory would be different from ../nist.

Reply:
I just didn't see much value in the abstraction group/mod provided. Plus it carries two unnecessary fields.